tools/ci_build/build.py

```diff
@@ -95,14 +100,21 @@ def parse_arguments():
     parser.add_argument("--use_llvm", action="store_true", help="Build tvm with llvm")
     parser.add_argument("--enable_msinternal", action="store_true", help="Enable for Microsoft internal builds only.")
     parser.add_argument("--llvm_path", help="Path to llvm dir")
-    parser.add_argument("--azure_sas_key", help="azure storage sas key, starts with '?'")
+    parser.add_argument("--azure_sas_key", help="Azure storage sas key, starts with '?'")
     parser.add_argument("--use_brainslice", action="store_true", help="Build with brain slice")
-    parser.add_argument("--brain_slice_package_path", help="Path to brain slice pacakges")
-    parser.add_argument("--brain_slice_package_name", help="Name of brain slice pakcages")
+    parser.add_argument("--brain_slice_package_path", help="Path to brain slice packages")
+    parser.add_argument("--brain_slice_package_name", help="Name of brain slice packages")
     parser.add_argument("--brain_slice_client_package_name", help="Name of brainslice client package")
     parser.add_argument("--use_nuphar", action='store_true', help="Build with nuphar")
     return parser.parse_args()
 
+def resolve_executable_path(command_or_path):
+    """Returns the absolute path of an executable."""
+    executable_path = shutil.which(command_or_path)
```

Comment:
Why adding this function? What if the input is already an absolute path?

Reply:
If it's already an absolute path it should be fine. If not, it may not work since the current directory gets changed for some commands. I installed a later cmake version on Linux and gave a relative path from the repo and that didn't work.

Reply:
Then why don't you give an absolute path to this script? People often have multiple cmake installed, calling `which cmake` at here may get unexpected result?

Reply:
Giving an absolute path works (which is what I had to do). It seemed like a bug that a relative path does not also work.

For --cmake_path, the default is "cmake". So for the default, I assumed running a cmake command starting with the result of shutil.which("cmake") would be equivalent.